apollo-router/src/plugins/telemetry/formatters/json.rs

```diff
@@ -0,0 +1,183 @@
+use std::collections::BTreeMap;
+use std::fmt;
+use std::fmt::Write;
+use std::io;
+
+use serde::ser::SerializeMap;
+use serde::ser::Serializer as _;
+use serde_json::Serializer;
+use tracing::span::Record;
+use tracing_core::Field;
+use tracing_subscriber::field;
+use tracing_subscriber::field::VisitOutput;
+use tracing_subscriber::fmt::format::Writer;
+use tracing_subscriber::fmt::FormatFields;
+use tracing_subscriber::fmt::FormattedFields;
+
+use super::TRACE_ID_FIELD_NAME;
+
+/// The JSON [`FormatFields`] implementation.
+///
+#[derive(Debug)]
+pub(crate) struct JsonFields;
+
+impl JsonFields {
+    /// Returns a new JSON [`FormatFields`] implementation.
+    ///
+    pub(crate) fn new() -> Self {
+        Self {}
+    }
+}
+
+impl Default for JsonFields {
+    fn default() -> Self {
+        Self::new()
+    }
+}
+
+impl<'a> FormatFields<'a> for JsonFields {
+    /// Format the provided `fields` to the provided `writer`, returning a result.
+    fn format_fields<R: field::RecordFields>(
+        &self,
+        mut writer: Writer<'_>,
+        fields: R,
+    ) -> fmt::Result {
+        let mut v = JsonVisitor::new(&mut writer);
+        fields.record(&mut v);
+        v.finish()
+    }
+
+    /// Record additional field(s) on an existing span.
+    ///
+    /// By default, this appends a space to the current set of fields if it is
+    /// non-empty, and then calls `self.format_fields`. If different behavior is
+    /// required, the default implementation of this method can be overridden.
```

Comment:
I don't think this comment is correct. Isn't this the comment from the plain text formatter?

Reply:
Let me check, I copy pasted some parts of the tracing code

Reply:
It looks like they copied that comment from the interface and it's wrong (I think) there as well. Never mind.